src/index.js

```diff
@@ -25,67 +27,78 @@ const port = +process.env.PORT || 3050;
+          hostname: process.env.PULSE_HOSTNAME,
+          vhost: process.env.PULSE_VHOST,
+        },
+      }),
+  },
+
+  schema: {
+    requires: [],
+    setup: () =>
+      createSchema({
+        typeDefs,
+        resolvers,
+        resolverValidationOptions: {
+          requireResolversForResolveType: false,
+        },
+        validationRules: [depthLimit(10), createComplexityLimitRule(1000)],
+      }),
   },
-});
-let schema;
-let context;
-let server;
-let app;
-let subscriptionServer;
-const load = () => {
-  schema = createSchema({
-    typeDefs,
-    resolvers,
-    resolverValidationOptions: {
-      requireResolversForResolveType: false,
-    },
-    validationRules: [depthLimit(10), createComplexityLimitRule(1000)],
-  });
-  context = createContext({ pulseEngine });
 
-  const express = servers({ schema, context, server });
+  context: {
+    requires: ['pulseEngine'],
+    setup: ({ pulseEngine }) => createContext({ pulseEngine }),
+  },
 
-  server = express.server; // eslint-disable-line prefer-destructuring
-  app = express.app; // eslint-disable-line prefer-destructuring
-  subscriptionServer = createSubscriptionServer({
-    server,
-    schema,
-    context,
-    path: '/subscription',
-  });
-};
+  app: {
+    requires: ['context', 'schema'],
+    setup: ({ context, schema }) => createApp({ schema, context }),
+  },
 
-load();
+  server: {
+    requires: ['app', 'schema', 'context'],
+    setup: ({ app, schema, context }) => {
+      const server = httpServer(app);
 
-server.listen(port, () => {
-  /* eslint-disable no-console */
-  console.log(`\n\nWeb server running on port ${port}.`);
-  console.log(
-    `\nOpen the interactive GraphQL Playground and schema explorer in your browser at:
-    http://localhost:${port}/playground\n`
-  );
-  /* eslint-enable no-console */
+      createSubscriptionServer({
+        server, // this attaches itself directly to the server
+        schema,
+        context,
+        path: '/subscription',
+      });
+
+      return server;
+    },
+  },
+
+  devServer: {
+    requires: ['server'],
+    setup: async ({ server }) => {
+      await new Promise(resolve => server.listen(port, resolve));
+
+      /* eslint-disable no-console */
+      console.log(`\n\nWeb server running on port ${port}.`);
+      console.log(
+        `\nOpen the interactive GraphQL Playground and schema explorer in your browser at:
+        http://localhost:${port}/playground\n`
+      );
+      /* eslint-enable no-console */
+    },
+  },
 });
 
-if (module.hot) {
-  module.hot.accept(
-    [
-      './resolvers',
-      './graphql',
-      './servers/createSubscriptionServer',
-      './servers',
-    ],
-    () => {
-      subscriptionServer.close();
-      server.removeListener('request', app);
-      load();
-      server.on('request', app);
-    }
-  );
+if (!module.parent) {
+  load('devServer').catch(err => {
+    console.log(err.stack); // eslint-disable-line no-console
+    process.exit(1);
+  });
```

Comment:
Should we include a default export in case the file has a parent?

Reply:
That will be useful if we add tests -- but we can add the export at that time :)